### PR TITLE
fix answers sorting functionality

### PR DESCRIPTION
**Pull request by s-cheng**
Currently sorting answers doesn't work at all. It seems like this was at least partly from the [giant "better stamp" PR](https://github.com/ForumMagnum/ForumMagnum/pull/7908), so doesn't have an explanation for the relevant changes made in there. This PR should fix it.

I think there is a more thorough version of a bug fix here but it's past 6 PM on Friday so I'm going to call it now. Maybe I'll look at this more next week.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1206743075210627) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/questions/AnswersSorting.tsx [hunk 1]
```diff
@@ -6,23 +6,15 @@ import * as _ from 'underscore';
 import type { Option } from '../common/InlineSelect';
 import { getCommentViewOptions } from '../../lib/commentViewOptions';
 import { useNavigate } from '../../lib/reactRouterWrapper';
-
```
Comment (s-cheng) [inline packages/lesswrong/components/questions/AnswersSorting.tsx:9]: The changes in this file basically revert the changes to this file in https://github.com/ForumMagnum/ForumMagnum/pull/7908 (while keeping the copy updates)

I think the change was an attempt to unify sorting between comments and answers, but comments seem to use different views to do the sorting while answers use different `sortBy` values passed into a single view.

File: packages/lesswrong/lib/collections/comments/views.ts [hunk 2]
```diff
@@ -462,16 +462,16 @@ Comments.addView('repliesToAnswer', (terms: CommentsViewTerms) => {
...
     selector: {
-      postId,
+      postId: terms.postId,
       $or: [
         { answer: true },
         { parentAnswerId: {$exists: true} },
       ],
     },
-    options: {sort: {baseScore: -1}}
+    options: {sort: questionAnswersSortings[terms.sortBy || "top"]}
```
Comment (s-cheng) [inline packages/lesswrong/lib/collections/comments/views.ts:474]: I don't know why this was missing, it seems like a separate bug which also made sorting answers broken? Anyway this seems reasonable to add

**Review by jpaddison3 — APPROVED** [review]
